shared/chat/conversation/list/container.js

```diff
@@ -0,0 +1,157 @@
+import type {Options} from '../messages'
+import type {Props, OptionsFn} from '.'
+import type {OwnProps, StateProps, DispatchProps} from './container'
+import type {TypedState} from '../../../constants/reducer'
+
+const mapStateToProps = (state: TypedState, {editLastMessageCounter, listScrollDownCounter, onFocusInput}: OwnProps): StateProps => {
+  const selectedConversationIDKey = Constants.getSelectedConversation(state)
+  const you = state.config.username || ''
+  const origFollowingMap = state.config.following
+  const origMetaDataMap = state.chat.get('metaData')
+
+  let firstNewMessageID = null
+  let followingMap = Map()
+  let messages = List()
+  let metaDataMap = Map()
+  let moreToLoad = false
+  let participants = List()
+  let editingMessage = null
+  let supersedes = null
+  let supersededBy = null
+  let validated = false
+
+  if (selectedConversationIDKey && Constants.isPendingConversationIDKey(selectedConversationIDKey)) {
+    const tlfName = Constants.pendingConversationIDKeyToTlfName(selectedConversationIDKey)
+    if (tlfName) {
+      participants = List(tlfName.split(','))
+      followingMap = pick(origFollowingMap, participants.toArray())
+      metaDataMap = origMetaDataMap.filter((k, v) => participants.contains(v))
+      validated = true
+    }
+  } else if (selectedConversationIDKey && selectedConversationIDKey !== Constants.nothingSelected) {
+    const conversationState = state.chat.get('conversationStates').get(selectedConversationIDKey)
+    if (conversationState) {
+      const inbox = state.chat.get('inbox')
+      const selected = inbox && inbox.find(inbox => inbox.get('conversationIDKey') === selectedConversationIDKey)
+
+      participants = selected && selected.participants || List()
+      firstNewMessageID = conversationState.firstNewMessageID
+      followingMap = pick(origFollowingMap, participants.toArray())
+      messages = conversationState.messages
+      metaDataMap = origMetaDataMap.filter((k, v) => participants.contains(v))
+      moreToLoad = conversationState.moreToLoad
+      editingMessage = state.chat.get('editingMessage')
+      supersedes = Constants.convSupersedesInfo(selectedConversationIDKey, state.chat)
+      supersededBy = Constants.convSupersededByInfo(selectedConversationIDKey, state.chat)
+      validated = selected && selected.state === 'unboxed'
+    }
+  }
+
+  return {
+    editLastMessageCounter,
+    editingMessage,
+    firstNewMessageID,
+    followingMap,
+    listScrollDownCounter,
+    messages,
+    metaDataMap,
+    moreToLoad,
+    onFocusInput,
+    participants,
+    selectedConversation: selectedConversationIDKey,
+    supersededBy,
+    supersedes,
+    validated,
+    you,
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
+  _onLoadAttachment: (selectedConversation, messageID, filename) => { dispatch(Creators.loadAttachment(selectedConversation, messageID, downloadFilePath(filename), false, false)) },
```

Comment:
flow really got confused by the pattern of overwriting a prop with a value of the same name. So i renamed these things to use _On... instead. I think this is also good cause it tells the devs these are meant to be overridden and not used directly